src/TimeModeling/Modeling/python_interface.jl

```diff
@@ -234,43 +238,47 @@ function devito_interface(modelPy::PyCall.PyObject, model, srcData::Array, recGe
 end
 
 # Jacobian of extended source modeling: d_lin = J*dm
-function devito_interface(modelPy::PyCall.PyObject, model, srcData::Array, recGeometry::Geometry, recData::Nothing, weights::Array,
+function devito_interface(modelPy::PyCall.PyObject, srcData::Array, recGeometry::Geometry, recData::Nothing, weights::Array,
                           dm::Union{PhysicalParameter, Array}, options::Options)
 
     # Interpolate input data to computational grid
-    dtComp = get_dt(model; dt=options.dt_comp)
-    qIn = time_resample(srcData,recGeometry,dtComp)[1]
+    dtComp = convert(Float32, modelPy."critical_dt")
+    qIn = time_resample(srcData, recGeometry, dtComp)[1]
 
     # Set up coordinates with devito dimensions
-    #origin = get_origin(modelPy)
     rec_coords = setup_grid(recGeometry, modelPy.shape)
 
     # Devito call
-    dOut = pycall(ac."born_rec_w", Array{Float32,2}, modelPy, weights, qIn, rec_coords,
-                  space_order=options.space_order, isic=options.isic)
-    dOut = time_resample(dOut,dtComp,recGeometry)
+    dOut = wrapcall_data(ac."born_rec_w", modelPy, weights, qIn, rec_coords,
+                         space_order=options.space_order, isic=options.isic)
+    dOut = time_resample(dOut, dtComp, recGeometry)
 
     # Output linearized shot records as judiVector
-    return judiVector{Float32, Array{Float32, 2}}("Je*dm", prod(size(dOut)), 1, 1, recGeometry, [dOut])
+    if options.save_data_to_disk
+        container = write_shot_record(srcGeometry,srcData,recGeometry,dOut,options)
```

Comment:
So I assume this used to be bug and we never found it before. Correct me if wrong

Reply:
I think it's never used since usually don't write to disc with `J` so just added it in case someone tries